src/js/importer.js

```diff
@@ -0,0 +1,374 @@
+//importer module
+
+'use strict';
+
+var fs = require('fs'),
+    path = require('path'),
+    request = require('request'),
+    AdmZip = require('adm-zip');
+
+function Importer(configurator, pm) {
+
+    return {
+
+        /**
+         * Imports a zip file containing txt files in USFM format or a single txt file in USFM format
+         * @param file {File} the path to the archive or txt file
+         * @returns {Promise.<boolean>}
+         */
+        importUSFMFile: function (file, translation, user) {
+            //console.log("importing USFM", JSON.stringify(file), "to project", JSON.stringify(translation));
+            var self = this;
+            let fileExt = file.name.split('.').pop().toLowerCase();
+            //console.log("File Extension: ", fileExt);
+            if (fileExt === "zip") {
+                return self.getTxtFilesFromZip(file).then(function (files) {
+                    //console.log("got files", files);
+                    let txtFile = files[0];
+                    var promise = self.importSingleUSFMFile(txtFile,translation,user);
+                    for (var i = 1; i < files.length; i++) {
+                        txtFile = files[i];
+                        promise = promise.then(function(){
+                            return self.importSingleUSFMFile(txtFile,translation,user);
+                        });
+                    }
+                    return promise;
+                });
+            } else {
+                //console.log("running single file import");
+                return self.importSingleUSFMFile(file.path,translation,user);
+            }
+        },
+
+        getTxtFilesFromZip: function (zipFile) {
+            var tempDir = path.join(configurator.getValue('tempDir'), "usfm-import");
+
+            return new Promise(function (resolve, reject) {
+                let zip = new AdmZip(zipFile.path);
+                zip.extractAllTo(tempDir, true);
+                var files = fs.readdirSync(tempDir),
+                    output = [];
+                files.forEach(function (file) {
+                    output.push(path.join(tempDir, file));
+                });
+
+                resolve(output);
+            });
+        },
+
+        getVerseChunkFileNames: function(projdata){
+            return new Promise(function(resolve,reject){
+                var book = projdata.project.id;
+                var chunkDescUrl = "https://api.unfoldingword.org/bible/txt/1/" + book + "/chunks.json";
+                var chunks = [];
+                request(chunkDescUrl,function(err, resp, body){
```

Comment:
Just noting that we'll bundle this into the app. This will become available in the index soon, for now you can leave this.